pkg/interop/client.go

```diff
@@ -0,0 +1,136 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package interop
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"reflect"
+	"strings"
+	"time"
+
+	"go.thethings.network/lorawan-stack/pkg/encoding/lorawan"
+	"go.thethings.network/lorawan-stack/pkg/ttnpb"
+	"go.thethings.network/lorawan-stack/pkg/types"
+)
+
+const (
+	// loRaAllianceDomain is the domain of LoRa Alliance.
+	loRaAllianceDomain = "lora-alliance.org"
+
+	// loRaAllianceJoinEUIDomain is the LoRa Alliance domain used for JoinEUI resolution.
+	loRaAllianceJoinEUIDomain = "joineuis." + loRaAllianceDomain
+
+	// loRaAllianceNetIDDomain is the LoRa Alliance domain used for NetID resolution.
+	loRaAllianceNetIDDomain = "netids." + loRaAllianceDomain
```

Comment:
Let's make these configurable on the cluster level with these values as default (no need for `const`)

Reply:
This is currently fully configurable, since e.g. `HandleJoinRequest` takes a `fqdn string`, which is turn can be generated by using `LoRaAllianceJoinServerFQDN`